src/SSA_stepper.jl

```diff
@@ -258,10 +258,15 @@ function DiffEqBase.step!(integrator::SSAIntegrator)
     # FP error means the new time may equal the old if the next jump time is
     # sufficiently small, hence we add this check to execute jumps until
     # this is no longer true.
+    integrator.u_modified = true
     while integrator.t == integrator.tstop
         doaffect && integrator.cb.affect!(integrator)
     end
 
+    if !integrator.u_modified
+        return nothing
+    end
+
```

Comment:
That's a good option which will ensure that we will only save when a jump occurs and still call the user callbacks. 

But then it doesn't save at every time the step of the stepper. From the discussion above that would be a desired behavior.

What if we added a new field to `SSAIntegrator` called `save_positions` and modified your proposal as following?

```julia
!saved_in_cb && savevalues!(integrator, jump_modified && integrator.save_positions)
```

Of course, I  believe we would also need to modify the `__init` to take the `save_everystep` keyword argument. However, I'm not so sure whether it is possible and how to do it. It seems to have a fixed number of keyword arguments and I can't trace the calling function back.

Reply:
I'm a little confused. My concerns were about not calling the callbacks every time step. It is fine with me if we only save jumps that change the state (since we would still have that `sol(t)` gives the exact sampled trajectory at every time `0 <= t < = tstop`). 

Shouldn't what I proposed preserve the current saving behavior for all non-coevolve methods, and save all jumps that change the state for coevolve?

Reply:
Your proposal would indeed address both not calling the callback every time step and it would only save the jumps that change the state.

What I had understood earlier is that you would like `save_everystep` to save every single step of the stepper which in my opinion might be more consistent.

I feel it is a bit confusing to have `save_positions` and `save_everystep` doing basically the same thing. The `__init` doesn't even use the argument `save_everystep` passed by the user. So `save_everystep = true` and `save_positions=(false, false)` won't save every jump as one would expect.

That's why I would advocate for treating them separately as well as to add a good documentation on their combined use. In fact, that was major pain point when I wrote the Synapse benchmark.

Reply:
There is no `save_everystep` keyword though, it isn't supported by `SSAStepper` and never has been. That is only a control for ODE/SDE timesteppers. Per step jump saving control is always via `save_positions`. But we should definitely add an error message if someone tries to use `save_everystep`, and better clarify the docs on saving controls with `SSAStepper`.

I propose we just accept the convention for now that non-jump timesteps never get saved in `SSAStepper`. I don't see how it could have an impact on correctness here since the exact path is still reconstructed via `sol` without knowing these times. If someone really wants to save the time of every candidate jump they can easily add a callback that saves every timestep.

We can then open an issue to discuss improving the `SSAStepper` saving controls, and map out how the various kwargs should interact. Then we can have a followup PR to fix them if they can be modified in a non-breaking way (or a strong argument can be made why we should have a breaking change).

This way we can get your PR merged, people can use it, and you aren't being held back with regards to JuliaCon. (And I can then focus on finishing going through the proceedings paper, which I've got a bunch of comments on in the half I've read so far...)